bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoApi.java

```diff
@@ -469,12 +512,16 @@ public NeeoDeviceKeys getDeviceKeys() {
      */
     public void notify(String msg) throws IOException {
         if (isConnected()) {
-            final HttpRequest rqst = request.get();
+            HttpClient httpClient = getHttpClient();
+            final HttpRequest rqst = new HttpRequest(httpClient);
             logger.debug("Sending Notification to brain ({}): {}", brainId, msg);
             final HttpResponse resp = rqst.sendPostJsonCommand(brainUrl + NeeoConstants.NOTIFICATION, msg);
             if (resp.getHttpCode() != HttpStatus.OK_200) {
                 throw resp.createException();
```

Comment:
In that case, `returnHttpClient(httpClient)` is not called. Is it ok ?
My feeling is that it will not put in the httpClientStack and so not stopped at the end.

Reply:
Valid point.  We should probably add a returnHttpClient prior to throwing the exception.

Reply:
Commit pushed to add.